extensions/cache/runtime/src/main/java/io/quarkus/cache/runtime/CacheResultInterceptor.java

```diff
@@ -53,35 +53,20 @@ public Object intercept(InvocationContext invocationContext) throws Throwable {
         try {
             ReturnType returnType = determineReturnType(invocationContext.getMethod().getReturnType());
             if (returnType != ReturnType.NonAsync) {
-                Uni<Object> cacheValue = cache.get(key, new Function<Object, Object>() {
+                Uni<Object> cacheValue = cache.getAsync(key, new Function<Object, Uni<Object>>() {
+                    @SuppressWarnings("unchecked")
                     @Override
-                    public Object apply(Object k) {
-                        LOGGER.debugf("Adding %s entry with key [%s] into cache [%s]",
-                                UnresolvedUniValue.class.getSimpleName(), key, binding.cacheName());
-                        return UnresolvedUniValue.INSTANCE;
-                    }
-                }).onItem().transformToUni(new Function<Object, Uni<?>>() {
-                    @Override
-                    public Uni<?> apply(Object value) {
-                        if (value == UnresolvedUniValue.INSTANCE) {
-                            try {
-                                return asyncInvocationResultToUni(invocationContext.proceed(), returnType)
-                                        .call(new Function<Object, Uni<?>>() {
-                                            @Override
-                                            public Uni<?> apply(Object emittedValue) {
-                                                return cache.replaceUniValue(key, emittedValue);
```

Comment:
`AbstractCache#replaceUniValue` and all its implementations are no longer used. They could be removed in this PR.